aws/resource_aws_ssm_parameter.go

```diff
@@ -105,30 +105,20 @@ func resourceAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("value", param.Value)
 
 	describeParamsInput := &ssm.DescribeParametersInput{
-		Filters: []*ssm.ParametersFilter{
-			&ssm.ParametersFilter{
+		ParameterFilters: []*ssm.ParameterStringFilter{
+			&ssm.ParameterStringFilter{
 				Key:    aws.String("Name"),
+				Option: aws.String("Equals"),
 				Values: []*string{aws.String(d.Get("name").(string))},
 			},
 		},
-		MaxResults: aws.Int64(50),
 	}
-	detailedParameters := []*ssm.ParameterMetadata{}
-	err = ssmconn.DescribeParametersPages(describeParamsInput,
-		func(page *ssm.DescribeParametersOutput, lastPage bool) bool {
-			detailedParameters = append(detailedParameters, page.Parameters...)
-			return !lastPage
-		})
+	describeResp, err := ssmconn.DescribeParameters(describeParamsInput)
 	if err != nil {
 		return fmt.Errorf("error describing SSM parameter: %s", err)
 	}
-	if len(detailedParameters) == 0 {
-		log.Printf("[WARN] SSM Param %q not found, removing from state", d.Id())
-		d.SetId("")
-		return nil
-	}
 
-	detail := detailedParameters[0]
+	detail := describeResp.Parameters[0]
```

Comment:
To prevent potential panics, we should perform a `nil` check on `describeResp` and `len()` check on `describeResp.Parameters` before attempting to reference, e.g.

```go
if describeResp == nil || len(describeResp.Parameters) == 0 || describeResp.Parameters[0] == nil {
  log.Printf("[WARN] SSM Parameter %q not found, removing from state", d.Id())
  d.SetId("")
  return nil
}
detail := describeResp.Parameters[0]
```